Parser/asdl_c.py

```diff
@@ -71,6 +71,21 @@ def is_simple(sum):
             return False
     return True
 
+def asdl_of(name, obj):
+    if isinstance(obj, asdl.Product) or isinstance(obj, asdl.Constructor):
+        fields = ", ".join(map(str, obj.fields))
+        if fields:
+            fields = "({})".format(fields)
+        return "{}{}".format(name, fields)
+    else:
+        if is_simple(obj):
+            types = " | ".join(type.name for type in obj.types)
+        else:
+            template = "\\n{}| ".format(" " * (len(name) + 3))
+            types = template.join(
+                asdl_of(type.name, type) for type in obj.types
```

Comment:
I hope all strings returned by nested `asdl_of()` are single-line. But for sure I would add `.replace('\n', '\n' + ' ' * (len(name) + 3))`.

Reply:
The only case we are returning something that isn't a single line is complex sums. And I couldn't figure out a way to put a complex sum inside of another complex sum. Can you give an example?

Reply:
I do not have example.